generator/scripts/gen-traffic.js

```diff
@@ -57,6 +57,35 @@ async function uploadRandomBytes(beePair, seed = 500, bytes = 1024 * 4 * 400) {
   console.log(`${beePair.uploaderBee.url} uploaded ${bytes} bytes to ${reference}`)
 }
 
+const DEFAULT_POLLING_FREQUENCY = 1_000
+const DEFAULT_STAMP_USABLE_TIMEOUT = 120_000
+
+async function waitUntilStampUsable(batchId, beeDebug, options = {}) {
+  const timeout = options.timeout || DEFAULT_STAMP_USABLE_TIMEOUT
+  const pollingFrequency = options.pollingFrequency || DEFAULT_POLLING_FREQUENCY
+  let timeoutReached = false
+
+  const timeoutPromise = async () =>
+    new Promise((_, reject) =>
+      setTimeout(() => {
+        timeoutReached = true
+        reject(new Error('Wait until stamp usable timeout has been reached'))
+      }, timeout),
+    )
+
+  const stampWaitPromise = async () => {
+    while (!timeoutReached) {
+      const stamp = await beeDebug.getPostageBatch(batchId)
+
+      if (stamp.usable) return stamp
+      await sleep(pollingFrequency)
+    }
+  }
+
+  // The typecasting is needed because technically stampWaitPromise can return undefined but the timeoutPromise would already throw exception
+  return Promise.race([stampWaitPromise(), timeoutPromise()])
+}
+
```

Comment:
I am fine with this (we use it in gateway-proxy) but thanks to @Cafe137 we also have a slightly less complex solution :) https://github.com/ethersphere/bee-dashboard/blob/master/src/utils/index.ts#L229-L245